src/functions/meta-assistant/send-daily-message/handler.ts

```diff
@@ -16,52 +12,72 @@ import Auth from "src/meta-assistant/auth/auth-provider";
  */
 export const sendDailyMessageHandler = async (): Promise<DailyHandlerResponse> => {
   try {
-    const { accessToken } = await Auth.getAccessToken();
-    if (!accessToken) {
-      throw new Error("Timebank authentication failed");
+    const severaApi = CreateSeveraApiService();
+    const severaUsers = await severaApi.getOptInUsers();
+    const previousWorkDays = TimeUtilities.getPreviousTwoWorkdays();
+    const workHours = await severaApi.getPreviousWorkHours();
+    const slackUsers = await SlackUtilities.getSlackUsers();
+    if (!severaUsers) {
+      throw new Error("No persons retrieved from Severa");
     }
 
-    const previousWorkDays = TimeUtilities.getPreviousTwoWorkdays();
-    const { yesterday, dayBeforeYesterday } = previousWorkDays;
+    // Get the user's guid from workHours
+    const workHoursUserGuids = Array.isArray(workHours) ? workHours.map(hour => hour.user.guid) : [];
 
-    const timebankUsers = await TimeBankApiProvider.getTimebankUsers(accessToken);
-    const slackUsers = await SlackUtilities.getSlackUsers();
-    const timeRegistrations = await ForecastApiUtilities.getTimeRegistrations(dayBeforeYesterday);
+    // Filter severaUsers to include only those found in WorkHours
+    const filteredSeveraUsers = Array.isArray(severaUsers) ? severaUsers.filter(user => workHoursUserGuids.includes(user.guid)) : [];
```

Comment:
Similar to above comment:
Using Array.isArray should not be necessary here, it is typed as SeveraResponseUsers returned from the api service but shouldn't it be typed as SeveraResponseUsers[]